### ISSUE TITLE
When training DiT/XL2 ， Which GPU do u used？

### ISSUE THREAD

**Issue by heepengpeng** [facebookresearch/DiT#22]
I use 8*RTX 3090（16G）， batchsize=256， cuda memory not enough

**Comment by wpeebles** [comment]
Yeah, I've trained it on an 80gb A100 node (8x GPUs). In order to fit on 3090s, you'll probably need to use some tricks like [Flash Attention](https://github.com/HazyResearch/flash-attention), AMP or gradient checkpointing.

**Comment by heepengpeng** [comment]
> For the PyTorch training script I've been using an 80gb A100 node (8x GPUs). In order to fit on 8x 3090s, you'll probably need to use some tricks like [Flash Attention](https://github.com/HazyResearch/flash-attention), AMP or gradient checkpointing.

how long when u you  use 80gb A100 node to train

**Comment by wpeebles** [comment]
Out of the box `train.py` runs at ~0.51 steps/sec on the A100 node. Lots of room for training throughput to be optimized though :)

**Comment by wpeebles** [comment]
Adding the following two lines at the top of `train.py` significantly increases speed from 0.51 steps/sec to 0.81 steps/sec on A100s:

```python
torch.backends.cuda.matmul.allow_tf32 = True
torch.backends.cudnn.allow_tf32 = True
```

Using `tf32` may change training results (I only tried training without `tf32`), but probably won't be very significant.
